d2graph/d2graph.go

```diff
@@ -1475,16 +1510,9 @@ func (g *Graph) SetDimensions(mtexts []*d2target.MText, ruler *textmeasure.Ruler
 func (g *Graph) Texts() []*d2target.MText {
 	var texts []*d2target.MText
 
-	capsLock := g.Theme != nil && g.Theme.SpecialRules.CapsLock
-
 	for _, obj := range g.Objects {
 		if obj.Attributes.Label.Value != "" {
 			text := obj.Text()
-			if capsLock && !strings.EqualFold(obj.Attributes.Shape.Value, d2target.ShapeCode) {
-				if obj.Attributes.Language != "latex" {
-					text.Text = strings.ToUpper(text.Text)
-				}
-			}
```

Comment:
how come these are removed?

Reply:
We are altering the `obj.Attributes.Label.Value` when we call the `ApplyTextTransform` method, which in turn means we no longer need to call `ToUpper` on it here.

Reply:
this function is not called in the course of compilation. it's just an exposed API for getting all the texts, so they still have to be applied here.

Reply:
Ok. I thought this would be covered in a test; so when I saw all text still looked as expected I thought it was safe to RM.

I will make the required changes to keep the if clause and add the other transformations.

Reply:
yeah there probably should be tests for it